simson/common/data_transformations.py

```diff
@@ -7,3 +9,60 @@ def broadcast_trailing_dimensions(array: np.ndarray, to_shape_of: np.ndarray) ->
     b_reshaped = np.reshape(array, new_shape)
     b_broadcast = np.broadcast_to(b_reshaped, to_shape_of.shape)
     return b_broadcast
+
+
+class Bound:
+    """Class representing bounds for a single parameter."""
+
+    def __init__(
+        self,
+        var_name: str,
+        lower_bound: Union[float, np.ndarray],
+        upper_bound: Union[float, np.ndarray],
+    ):
+        self.var_name = var_name
+        self.lower_bound = np.array(lower_bound)
+        self.upper_bound = np.array(upper_bound)
+
+    @staticmethod
+    def create_bounds_arr(
+        bounds_list: list["Bound"], all_prm_names: list[str], bound_shape: tuple
+    ) -> np.ndarray:
+        """Creates bounds array where each element is tuple of lower and upper bounds for each parameter."""
+
+        # Check bound shapes
+        if bounds_list:
```

Comment:
I would make this more explicit
```suggestion
        if isinstance(bounds_list, iterable) and len(bounds_list) >= 1:
```